src/hooks/useCookieState.ts

```diff
@@ -0,0 +1,44 @@
+import Cookies from 'js-cookie';
+import { useCallback, useState } from 'react';
+
+type UseCookieStateType = {
+  value: string | null;
+  updateCookie: (newValue: string) => void;
+  deleteCookie: () => void;
+} | null;
+
+/**
+ *
+ * @param cookieName Name of the cookie
+ * @returns {handler} A handler to interact with the cookie
+ */
+const useCookieState = (
+  cookieName: string,
+  cookieValue: string | null = null,
+  options?: Cookies.CookieAttributes
+): UseCookieStateType => {
+  const [value, setValue] = useState<string | null>(() => {
+    if (typeof window !== undefined) {
+      return Cookies.get(cookieName, options) || cookieValue;
+    }
+
+    return null;
+  });
+
```

Comment:
We need useCookieState to listen for updates to this cookie from external events. So basically a listener on cookie change similar to how our useLocalstorageState works. WE need something for cookies .